iontestdata/good/typecodes/README.md

```diff
@@ -0,0 +1,1153 @@
+> 10 byte decimal  
+> _-2.361183241434822606847d-42_ or _-2.361183241434822606847E-42_
+
+> `5B FF FF FF FF FF FF FF FF FF FF FF`  
+> 11 byte decimal  
+> _-6.04462909807314587353087d-40_ or _-6.04462909807314587353087E-40_
+
+> `5C FF FF FF FF FF FF FF FF FF FF FF FF`  
+> 12 byte decimal  
+> _-1.54742504910672534362390527d-37_ or _-1.54742504910672534362390527E-37_
+
+> `5D FF FF FF FF FF FF FF FF FF FF FF FF FF`  
+> 13 byte decimal  
+> _-3.9614081257132168796771975167d-35_ or _-3.9614081257132168796771975167E-35_
+
+> `5E 8E FF FF FF FF FF FF FF FF FF FF FF FF FF FF`  
+> VarUInt (8E = 14) byte decimal  
+> _-1.0141204801825835211973625643007d-32_ or _-1.0141204801825835211973625643007E-32_
+
+> `5F`  
+> _null.decimal_
+
+### Invalid decimal type descriptors
+
+> _none_
+
+## T6 - timestamp
+
+Specification: http://amzn.github.io/ion-docs/docs/binary.html#6-timestamp  
+
+```
+E0 01 00 EA 62 E1 E1 63 E1 E1 81 64 E1 E1 81 81 
+65 E1 12 E1 81 81 66 E1 E1 81 81 81 81 67 E1 E1 
+81 81 81 81 81 68 E1 E1 81 81 81 81 81 E1 69 E1 
+E1 81 81 81 81 81 E1 12 6A E1 E1 81 81 81 81 81 
+E1 12 12 6B E1 E1 81 81 81 81 81 E1 12 12 12 6C 
+E1 E1 81 81 81 81 81 E1 12 12 12 12 6D E1 E1 81 
+81 81 81 81 E1 12 12 12 12 12 6E 8E E1 E1 81 81 
+81 81 81 E1 12 12 12 12 12 12 6F  
+ 
+```
+
+> `E0 01 00 EA`  
+> Binary Version Marker (BVM)
+
+> `62 E1 E1`  
+> 2 byte timestamp with at most VarInt offset and VarUInt year components  
+> month, day, hour, minute, second, fraction_exponent, and fraction_coefficient components are not able to be specified  
+> _0096T_
+
+> `63 E1 E1 81`  
+> 3 byte timestamp with at most VarInt offset, VarUInt year, and VarUInt month components  
+> day, hour, minute, second, fraction_exponent, and fraction_coefficient components are not able to be specified  
+> _0096-12T_
+
+> `64 E1 E1 81 81`  
+> 4 byte timestamp with at most VarInt offset, VarUInt year, VarUInt month, and VarUInt day components  
+> hour, minute, second, fraction_exponent, and fraction_coefficient components are not able to be specified  
+> _0096-12-31T_
+
+> `65 E1 12 E1 81 81`  
+> 5 byte timestamp with at most VarInt offset, VarUInt year, VarUInt month, and VarUInt day components  
+> hour, minute, second, fraction_exponent, and fraction_coefficient components are not able to be specified  
+> _2400-12-31T_
+
+> `66 E1 E1 81 81 81 81`  
+> 6 byte timestamp with at most VarInt offset, VarUInt year, VarUInt month, VarUInt day, VarUInt hour, and VarUInt minute components  
+> second, fraction_exponent, and fraction_coefficient components are not able to be specified  
+> _0097-01-01T00:28-00:33_
+
+> `67 E1 E1 81 81 81 81 81`  
+> 7 byte timestamp with at most VarInt offset, VarUInt year, VarUInt month, VarUInt day, VarUInt hour, VarUInt minute, and VarUInt second  components  
+> fraction_exponent, and fraction_coefficient components are not able to be specified  
+> _0097-01-01T00:28:01-00:33_
+
+> `68 E1 E1 81 81 81 81 81 E1`  

```

Comment:
I recommend splitting the timestamps with large fractional precision into a separate file like you did with symbols. Some implementations have open issues for supporting arbitrary fractional precision.

Alternatively, you could try just using a lower precision that would work for all implementations, as we already have tests for high fractional precision.

Reply:
I split the file. Line 5 (now updated) attempts to explain the reasoning why I am sticking with a high fractional precision.